webroot/src/pages/SelectPrivileges/SelectPrivileges.ts

```diff
@@ -0,0 +1,380 @@
+            (formInput.value === true)).map((input) => input.id);
+    }
+
+    get selectedStatePurchaseDataList(): Array<PrivilegePurchaseOption> {
+        return this.purchaseDataList.filter((option) => {
+            let includes = false;
+
+            const stateAbbrev = option?.jurisdiction?.abbrev;
+
+            if (stateAbbrev) {
+                includes = this.statesSelected?.includes(stateAbbrev);
+            }
+
+            return includes;
+        });
+    }
+
+    get selectedStatesWithJurisprudenceRequired(): Array<string> {
+        const selectedStatesWithJurisprudenceRequired: Array<string> = [];
+        const privPurchaseStatesRequiringJurisprudence = this.selectedStatePurchaseDataList
+            .filter((state) => state.isJurisprudenceRequired);
+
+        privPurchaseStatesRequiringJurisprudence.forEach((state) => {
+            if (state?.jurisdiction?.abbrev) {
+                selectedStatesWithJurisprudenceRequired.push(state.jurisdiction.abbrev);
+            }
+        });
+
+        return selectedStatesWithJurisprudenceRequired;
+    }
+
+    get expirationDateText(): string {
+        return this.$t('licensing.expirationDate');
+    }
+
+    get jurisdictionFeeText(): string {
+        return this.$t('licensing.jurisdictionFee');
+    }
+
+    get commissionFeeText(): string {
+        return this.$t('licensing.commissionFee');
+    }
+
+    get subtotalText(): string {
+        return this.$t('common.subtotal');
+    }
+
+    get jurisprudenceExplanationText(): string {
+        return this.$t('licensing.jurisprudenceExplanationText');
+    }
+
+    get militaryDiscountText(): string {
+        return this.$t('licensing.militaryDiscountText');
+    }
+
+    get jurisprudenceModalTitle(): string {
+        return this.$t('licensing.jurisprudenceConfirmation');
+    }
+
+    get jurisprudenceModalContent(): string {
+        return this.$t('licensing.jurisprudenceUnderstandParagraph');
+    }
+
+    get iUnderstandText(): string {
+        return this.$t('licensing.iUnderstand');
+    }
+
+    get activeLicense(): License | null {
+        return this.licenseList?.find((license) => license.statusState === LicenseStatus.ACTIVE) || null;
+    }
+
+    get activeLicenseExpirationDate(): string {
+        let date = '';
+
+        if (this.activeLicense) {
+            const { expireDate } = this.activeLicense;
+
+            if (expireDate) {
+                date = moment(expireDate).format(displayDateFormat);
+            }
+        }
+
+        return date;
+    }
+
+    get subTotalList(): Array<number> {
+        return this.selectedStatePurchaseDataList?.map((purchaseInfo) => {
+            const militaryDiscount = purchaseInfo.isMilitaryDiscountActive && purchaseInfo.militaryDiscountAmount
+                ? purchaseInfo.militaryDiscountAmount : 0;
+
+            return ((purchaseInfo?.fee || 0) + (this.currentCompactCommissionFee || 0) - (militaryDiscount));
```

Comment:
This could in theory be negative, but I assume that's something we will just verify on the configuration end and not worth checking here?

Reply:
True; however to your point, I think that would be a bigger issue than displaying a negative number here. I think the front end just reporting what the backend gives it is what we want here. Unless the backend is handling this differently(minimum of zero etc) I dont think we want to display something different.

Reply:
Ok checked with Landon, he said it will throw an error so I think displaying the negative is good in that case because it will tip off that there is something weird and inform on the error to an extent

Reply:
Sounds like a great plan!